### PR TITLE
Fixes #16636: Purge of unreferenced software may fail on very large system

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/16636

### CODE REVIEW

File: webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/ReadOnlySoftwareInventoryDAOImpl.scala [hunk 1]
```diff
@@ -123,18 +123,24 @@ class ReadOnlySoftwareDAOImpl(
 
       // fetch all nodes
       nodes        = con.searchSub(nodeBaseSearch, IS(OC_NODE), A_NODE_UUID)
-      batchedNodes = nodes.grouped(50).toSeq
-
+      batchedNodes = nodes.grouped(25).toSeq
```
Comment (fanf) [inline webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/ReadOnlySoftwareInventoryDAOImpl.scala:126]: can you put the `25` in a var and use that var in log? I'm almost (to be fair with me) that I would forget half of the change if I ever touch that

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**